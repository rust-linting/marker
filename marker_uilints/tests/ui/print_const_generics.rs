struct PrintMeConstGenerics<const N: usize> {
    data: [f32; N],
}

fn print_me() -> PrintMeConstGenerics<3> {
    todo!()
}

impl<const N: usize> PrintMeConstGenerics<N> {}

fn main() {
    let _ty: PrintMeConstGenerics<3> = PrintMeConstGenerics { data: [1.0, 1.5, 2.0] };
}
