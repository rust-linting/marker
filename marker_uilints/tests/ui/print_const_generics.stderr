warning: printing item
 --> $DIR/print_const_generics.rs:1:8
  |
1 | struct PrintMeConstGenerics<const N: usize> {
  |        ^^^^^^^^^^^^^^^^^^^^
  |
  = note: Struct(
              StructItem {
                  data: CommonItemData {
                      id: ItemId(..),
                      span: SpanId(..),
                      vis: Visibility {
                          span: None,
                          sem: Visibility {
                              _lifetime: PhantomData<&()>,
                              kind: DefaultCrate(
                                  ItemId(..),
                              ),
                          },
                      },
                      ident: Ident {
                          name: "PrintMeConstGenerics",
                          span: $DIR/print_const_generics.rs:1:8 - 1:28,
                      },
                  },
                  generics: GenericParams {
                      params: [
                          Const(
                              ConstParam {
                                  id: GenericId(..),
                                  name: SymbolId(..),
                                  ty: Num(
                                      NumTy {
                                          data: CommonSynTyData {
                                              _lifetime: PhantomData<&()>,
                                              span: SpanId(..),
                                          },
                                          numeric_kind: Usize,
                                      },
                                  ),
                                  default: None,
                                  span: SpanId(..),
                              },
                          ),
                      ],
                      clauses: [],
                  },
                  kind: Field(
                      [
                          ItemField {
                              id: FieldId(..),
                              vis: Visibility {
                                  span: None,
                                  sem: Visibility {
                                      _lifetime: PhantomData<&()>,
                                      kind: DefaultCrate(
                                          ItemId(..),
                                      ),
                                  },
                              },
                              ident: SymbolId(..),
                              ty: Array(
                                  ArrayTy {
                                      data: CommonSynTyData {
                                          _lifetime: PhantomData<&()>,
                                          span: SpanId(..),
                                      },
                                      inner_ty: Num(
                                          NumTy {
                                              data: CommonSynTyData {
                                                  _lifetime: PhantomData<&()>,
                                                  span: SpanId(..),
                                              },
                                              numeric_kind: F32,
                                          },
                                      ),
                                      len: Some(
                                          ConstExpr {
                                              expr: Path(
                                                  PathExpr {
                                                      data: CommonExprData {
                                                          _lifetime: PhantomData<&()>,
                                                          id: ExprId(..),
                                                          span: SpanId(..),
                                                      },
                                                      path: AstQPath {
                                                          self_ty: None,
                                                          path_ty: None,
                                                          path: AstPath {
                                                              segments: [
                                                                  AstPathSegment {
                                                                      ident: Ident {
                                                                          name: "N",
                                                                          span: $DIR/print_const_generics.rs:2:17 - 2:18,
                                                                      },
                                                                      generics: GenericArgs {
                                                                          args: [],
                                                                      },
                                                                  },
                                                              ],
                                                          },
                                                          target: Generic(
                                                              GenericId(..),
                                                          ),
                                                      },
                                                  },
                                              ),
                                          },
                                      ),
                                  },
                              ),
                              span: SpanId(..),
                          },
                      ],
                  ),
              },
          )
  = note: `#[warn(marker::marker_uilints::test_lint)]` on by default

warning: printing item
 --> $DIR/print_const_generics.rs:5:4
  |
5 | fn print_me() -> PrintMeConstGenerics<3> {
  |    ^^^^^^^^
  |
  = note: Fn(
              FnItem {
                  data: CommonItemData {
                      id: ItemId(..),
                      span: SpanId(..),
                      vis: Visibility {
                          span: None,
                          sem: Visibility {
                              _lifetime: PhantomData<&()>,
                              kind: DefaultCrate(
                                  ItemId(..),
                              ),
                          },
                      },
                      ident: Ident {
                          name: "print_me",
                          span: $DIR/print_const_generics.rs:5:4 - 5:12,
                      },
                  },
                  generics: GenericParams {
                      params: [],
                      clauses: [],
                  },
                  constness: NotConst,
                  syncness: Sync,
                  safety: Safe,
                  is_extern: false,
                  has_self: false,
                  abi: Default,
                  params: [],
                  return_ty: Some(
                      Path(
                          PathTy {
                              data: CommonSynTyData {
                                  _lifetime: PhantomData<&()>,
                                  span: SpanId(..),
                              },
                              path: AstQPath {
                                  self_ty: None,
                                  path_ty: None,
                                  path: AstPath {
                                      segments: [
                                          AstPathSegment {
                                              ident: Ident {
                                                  name: "PrintMeConstGenerics",
                                                  span: $DIR/print_const_generics.rs:5:18 - 5:38,
                                              },
                                              generics: GenericArgs {
                                                  args: [
                                                      Const(
                                                          ConstArg {
                                                              span: SpanId(..),
                                                              expr: ConstExpr {
                                                                  expr: IntLit(
                                                                      IntLitExpr {
                                                                          data: CommonExprData {
                                                                              _lifetime: PhantomData<&()>,
                                                                              id: ExprId(..),
                                                                              span: SpanId(..),
                                                                          },
                                                                          value: 3,
                                                                          suffix: None,
                                                                      },
                                                                  ),
                                                              },
                                                          },
                                                      ),
                                                  ],
                                              },
                                          },
                                      ],
                                  },
                                  target: Item(
                                      ItemId(..),
                                  ),
                              },
                          },
                      ),
                  ),
                  body_id: Some(
                      BodyId(..),
                  ),
              },
          )

warning: print type test
  --> $DIR/print_const_generics.rs:12:5
   |
12 |     let _ty: PrintMeConstGenerics<3> = PrintMeConstGenerics { data: [1.0, 1.5, 2.0] };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: Adt(
               AdtTy {
                   data: CommonTyData {...},
                   def_id: TyDefId(..),
                   generics: GenericArgs {
                       args: [
                           Const(
                               ConstArg {
                                   value: ConstValue {{ /* WIP: See rust-marker/marker#179 */}},
                               },
                           ),
                       ],
                   },
               },
           )

warning: 3 warnings emitted

